internal/services/contracts_dcn_handler_test.go

```diff
@@ -0,0 +1,203 @@
+	_, addr, err := test.GenerateWallet()
+	o.NoError(err)
+
+	_, owner, err := test.GenerateWallet()
+	o.NoError(err)
+
+	var eventData = NewDCNResolverEventData{
+		Node:     []byte("0xc6e7df5e7b4f2a278906862b612058"),
+		Resolver: *addr,
+	}
+
+	config := mocks.NewTestConfig()
+	consumer := mocks.NewConsumer(o.T(), config)
+
+	d := models.DCN{
+		Node:         eventData.Node,
+		OwnerAddress: owner.Bytes(),
+	}
+
+	err = d.Insert(o.ctx, o.pdb.DBS().Writer.DB, boil.Infer())
+	o.NoError(err)
+
+	contractEventConsumer := NewContractsEventsConsumer(o.pdb, &o.logger, &o.settings)
+	expectedBytes := eventBytes(eventData, contractEventData, o.T())
+
+	consumer.ExpectConsumePartition(o.settings.ContractsEventTopic, 0, 0).YieldMessage(&sarama.ConsumerMessage{Value: expectedBytes})
+
+	outputTest, err := consumer.ConsumePartition(o.settings.ContractsEventTopic, 0, 0)
+	o.NoError(err)
+
+	m := <-outputTest.Messages()
+	var e shared.CloudEvent[json.RawMessage]
+	err = json.Unmarshal(m.Value, &e)
+	o.NoError(err)
+
+	err = contractEventConsumer.Process(o.ctx, &e)
+	o.NoError(err)
+
+	dcn, err := models.DCNS().All(o.ctx, o.pdb.DBS().Reader.DB)
+	o.NoError(err)
+
+	o.Len(dcn, 1)
+	o.Equal(eventData.Node, dcn[0].Node)
+	o.Equal(owner.Bytes(), dcn[0].OwnerAddress)
+	o.Equal(addr.Bytes(), dcn[0].ResolverAddress.Bytes)
+}
+
+func (o *DCNConsumerTestSuite) Test_NewDCNExpiration_Consume_Success() {
+	contractEventData.EventName = NewExpiration.String()
+	_, addr, err := test.GenerateWallet()
+	o.NoError(err)
+
+	_, owner, err := test.GenerateWallet()
+	o.NoError(err)
+
+	currTime := time.Now().UTC().Truncate(time.Second)
+
+	var eventData = NewDCNExpirationEventData{
+		Node:       []byte("0xc6e7df5e7b4f2a278906862b612058"),
+		Expiration: int(currTime.Unix()),
+	}
+
+	config := mocks.NewTestConfig()
+	consumer := mocks.NewConsumer(o.T(), config)
+
+	d := models.DCN{
+		Node:            eventData.Node,
+		OwnerAddress:    owner.Bytes(),
+		ResolverAddress: null.BytesFrom(addr.Bytes()),
+	}
+
+	err = d.Insert(o.ctx, o.pdb.DBS().Writer.DB, boil.Infer())
+	o.NoError(err)
+
+	contractEventConsumer := NewContractsEventsConsumer(o.pdb, &o.logger, &o.settings)
+	expectedBytes := eventBytes(eventData, contractEventData, o.T())
+
+	consumer.ExpectConsumePartition(o.settings.ContractsEventTopic, 0, 0).YieldMessage(&sarama.ConsumerMessage{Value: expectedBytes})
+
+	outputTest, err := consumer.ConsumePartition(o.settings.ContractsEventTopic, 0, 0)
+	o.NoError(err)
+
+	m := <-outputTest.Messages()
+	var e shared.CloudEvent[json.RawMessage]
+	err = json.Unmarshal(m.Value, &e)
+	o.NoError(err)
+
+	err = contractEventConsumer.Process(o.ctx, &e)
+	o.NoError(err)
+
+	dcn, err := models.DCNS().All(o.ctx, o.pdb.DBS().Reader.DB)
+	o.NoError(err)
+
+	o.Len(dcn, 1)
+	o.Equal(eventData.Node, dcn[0].Node)
+	o.Equal(owner.Bytes(), dcn[0].OwnerAddress)
+	o.Equal(addr.Bytes(), dcn[0].ResolverAddress.Bytes)
+	o.Equal(currTime, dcn[0].Expiration.Time)
+}
```

Comment:
I think one gap in these tests is that we’re not checking the serialization, and I suspect there’s something wrong there.